extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -558,25 +558,34 @@ export class AmpA4A extends AMP.BaseElement {
     // So long as any one of these signing services can verify the
     // signature, then the creative is valid AMP.
     const keyInfoSetPromises = this.win.ampA4aValidationKeys;
+    // Track if verification found as it will ensure that promises yet to
+    // resolve will "cancel" as soon as possible saving unnecessary resource
+    // allocation.
+    let verificationFound = false;
+    const alreadyVerified = () => {
+      return verificationFound ?
+          Promise.reject('verification already found') : null;
+    };
     return some(keyInfoSetPromises.map(keyInfoSetPromise => {
       // Resolve Promise into Array of Promises of signing keys.
       return keyInfoSetPromise.then(keyInfoSet => {
         // As long as any one individual key of a particular signing
         // service, keyInfoPromise, can verify the signature, then the
         // creative is valid AMP.
-        return some(keyInfoSet.map(keyInfoPromise => {
+        return !alreadyVerified() && some(keyInfoSet.map(keyInfoPromise => {
```

Comment:
Won't this drop the `Promise.reject()` on the floor when `verificationFound`?  `!Promise.reject('foo') == false`, so you'll go on to the `some` clause, but the `Promise` will be uncaptured and produce an uncaught promise exception, won't it?